skip.js

```diff
@@ -0,0 +1,236 @@
+var ShortBufferError = require('bufrw/errors').ShortBuffer;
+var InvalidTypeidError = require('./errors').InvalidTypeidError;
+var TMapHeaderRW = require('./tmap').TMapRW.prototype.headerRW;
+var TListHeaderRW = require('./tlist').TListRW.prototype.headerRW;
+
+var widths = Object.create(null);
+widths[TYPE.VOID] = 0;
+widths[TYPE.BOOL] = 1;
+widths[TYPE.BYTE] = 1;
+widths[TYPE.I8] = 1;
+widths[TYPE.DOUBLE] = 8;
+widths[TYPE.I16] = 2;
+widths[TYPE.I32] = 4;
+widths[TYPE.I64] = 8;
+
+var skipVar = Object.create(null);
+skipVar[TYPE.STRUCT] = skipStruct;
+skipVar[TYPE.STRING] = skipString;
+skipVar[TYPE.MAP] = skipMap;
+skipVar[TYPE.SET] = skipList;
+skipVar[TYPE.LIST] = skipList;
+
+function skipField(buffer, offset) {
+
+    // istanbul ignore if
+    if (offset + 1 > buffer.length) {
+        return new ReadResult(ShortBufferError({
+            expected: offset + 1,
+            actual: buffer.length,
+            buffer: buffer,
+            offset: offset
+        }));
+    }
+
+    var typeid = buffer.readInt8(offset, true);
+    offset += 1;
+
+    return skipType(buffer, offset, typeid);
+}
+
+function skipType(buffer, offset, typeid) {
+    var result;
+
+    // istanbul ignore else
+    if (skipVar[typeid] !== undefined) {
+        result = skipVar[typeid](buffer, offset);
+        // istanbul ignore if
+        if (result.err) {
+            return result;
+        }
+        offset = result.offset;
+
+    } else if (widths[typeid] !== undefined) {
+        var length = widths[typeid];
+        if (offset + length > buffer.length) {
+            return new ReadResult(ShortBufferError({
+                expected: offset + length,
+                actual: buffer.length,
+                buffer: buffer,
+                offset: offset
+            }));
+        }
+        offset += length;
+
+    } else {
+        return new ReadResult(InvalidTypeidError({
+            typeid: typeid,
+            what: 'field::type'
+        }));
+    }
+
+    return new ReadResult(null, offset);
+}
+
+function skipStruct(buffer, offset) {
+    var result;
+    for (;;) {
+
+        // typeid
+        // istanbul ignore if
+        if (offset + 1 > buffer.length) {
+            return new ReadResult(ShortBufferError({
+                expected: offset + 1,
+                actual: buffer.length,
+                buffer: buffer,
+                offset: offset
+            }));
+        }
+        var typeid = buffer.readInt8(offset, true);
+        offset += 1;
+
+        if (typeid === TYPE.STOP) {
+            return new ReadResult(null, offset);
+        }
+
+        // id
+        // istanbul ignore if
+        if (offset + 2 > buffer.length) {
+            return new ReadResult(ShortBufferError({
+                expected: offset + 2,
+                actual: buffer.length,
+                buffer: buffer,
+                offset: offset
+            }));
+        }
+        offset += 2;
+
+        result = skipType(buffer, offset, typeid);
```

Comment:
suggest push down `var result` to here to avoid accidental (lint obvious) erroneous usage.